db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/PredicateList.java

```diff
@@ -2663,6 +2664,77 @@ private void generateSingleQualifierCode(MethodBuilder consMB,
         consMB.callMethod(VMOpcode.INVOKESTATIC,acb.getBaseClassName(),"setQualifier","void",4);
     }
 
+    /**
+     * generate ORed terms from inlist predicate
+     */
+    private void generateSingleQualifierCodeForInListElement(MethodBuilder consMB,
+                                             ExpressionClassBuilder acb,
+                                             int columnPosition,
+                                             int storagePosition,
+                                             ValueNode valNode,
+                                             LocalField qualField,
+                                             int array_idx_1,
+                                             int array_idx_2) throws StandardException{
+        consMB.getField(qualField); // first arg for setQualifier
+
+        // get instance for getQualifier call
+        consMB.pushThis();
+        consMB.callMethod(VMOpcode.INVOKEVIRTUAL,acb.getBaseClassName(),"getExecutionFactory",ExecutionFactory.MODULE,0);
+
+        // Column Id - first arg
+
+        consMB.push(columnPosition);
+        consMB.push(storagePosition);
+
+        // Operator - second arg
+        consMB.push(Orderable.ORDER_OP_EQUALS);
+
+        // Method to evaluate qualifier -- third arg
+        MethodBuilder qualMethod = acb.newUserExprFun();
+        valNode.generateExpression(acb, qualMethod);
+        qualMethod.methodReturn();
+        qualMethod.complete();
+        acb.pushMethodReference(consMB, qualMethod);
+
+        // Receiver for above method - fourth arg
+        acb.pushThisAsActivation(consMB);
+
+        // Ordered Nulls? - fifth arg
+        consMB.push(false);
+
+
+        /*
+        ** "Unknown" return value. For qualifiers,
+        ** we never want to return rows where the
+        ** result of a comparison is unknown.
+        ** But we can't just generate false, because
+        ** the comparison result could be negated.
+        ** So, generate the same as the negation
+        ** operand - that way, false will not be
+        ** negated, and true will be negated to false.
+        */
+        consMB.push(false);
+
+        /* Negate comparison result? */
+        consMB.push(false);
+
+        /* variantType for qualifier's orderable */
+        consMB.push(valNode.getOrderableVariantType());
+
+        int numArgs=10;
+        /* Qualifier's string representation */
```

Comment:
Not sure if we want to have a fixed string representation for any qualifiers. But we can change it later.